src/gobapi/storage.py

```diff
@@ -387,18 +391,29 @@ def get_entity(catalog, collection, id, view=None):
     return entity_convert(entity) if entity else None
 
 
+def filter_deleted(query, model):
+    """Filter a query to exclude records with date deleted
+
+    :param query:
+    :param model: The SQLAlchemy model
+    :return: query
+    """
+    # The table can also be a view on a table and doesn't always have _date_deleted
+    try:
+        query = query.filter(getattr(model, FIELD.DATE_DELETED) == None)  # noqa: E711 (== None)
+    except AttributeError:
+        pass
+    return query
+
```

Comment:
Does this not create an error? query is undefined when the exception is catched.
Probably it is more explicit to just check if the attribute is in the model.